functions/__fish_apple_touchbar_bind_key.fish

```diff
@@ -1,8 +1,12 @@
 function __fish_apple_touchbar_bind_key --argument-names fn_number fn_text fn_command bind_option
     __fish_apple_touchbar_print_key "SetKeyLabel=F$fn_number=$fn_text"
 
+    if bind -M insert >/dev/null 2>/dev/null
+        set vimbind "-M" "insert"
+    end
+
     if [ "$bind_option" = "-s" ]
-        bind --key f$fn_number "commandline -r '$fn_command'; commandline -f execute"
+        bind $vimbind --key f$fn_number "commandline -r '$fn_command'; commandline -f execute"
```

Comment:
Will this work even if `vimbind` is not set?

Reply:
yes, it just gets evaluated to an empty string (`""`), and should thus not affect anything.
But it would be great if you could check out that commit and try it locally - I am not 100% certain that the `bind -M insert` catches all cases.
(It didn't work for me when not calling `fish_vi_key_bindings`, but it could be that i have some plugins loaded that add their own vi bindings).